hibernate-core/src/main/java/org/hibernate/dialect/DB2zDialect.java

```diff
@@ -113,7 +113,11 @@ public SequenceSupport getSequenceSupport() {
 
 	@Override
 	public String getQuerySequencesString() {
-		return "select * from sysibm.syssequences";
+		return """
+				select '' as sequence_catalog, seqschema, seqname, start, minvalue, maxvalue, increment from sysibm.syssequences where seqtype='A'
+				union
+				select '' as sequence_catalog, schema as seqschema, name as seqname, start, minvalue, maxvalue, increment from sysibm.syssequences where seqtype!='A'
+				""";
```

Comment:
Since `SequenceInformationExtractorDB2DatabaseImpl` doesn't support catalogs, there is no point in selecting an empty string as catalog. Also, instead of a union query, it might be easier to just use a case expression.

```suggestion
		return "select case when seqtype='A' then seqschema else schema end as seqschema, case when seqtype='A' then seqname else name end as seqname, start, minvalue, maxvalue, increment from sysibm.syssequences";
```